kata_agent.go

```diff
@@ -295,6 +295,82 @@ func (k *kataAgent) exec(pod *Pod, c Container, cmd Cmd) (*Process, error) {
 	return prepareAndStartShim(pod, k.shim, c.id, req.ExecId, k.state.URL, cmd)
 }
 
+func (k *kataAgent) sendNetworkInterfaceAndRouteRequests(pod Pod) error {
+
+	networkNS, err := pod.storage.fetchPodNetwork(pod.id)
+	if err != nil {
+		return err
+	}
+
+	if networkNS.NetNsPath == "" {
+		return nil
+	}
+
+	for _, endpoint := range networkNS.Endpoints {
```

Comment:
Random thought: this code is pretty similar to:

- https://github.com/containers/virtcontainers/blob/master/hyperstart_agent.go#L173

I wonder if it would be worth refactoring to make a generic function in `agent.go` that accepted a `Pod` and a function to operate on the `Endpoint`'s?

Reply:
It is similar in that it loops over the endpoints and addresses, but with the protocols being different the implementation is different enough (operating on an interface and route at a time) that I think it may be a bit painful for a potentially short term code reuse benefit.  WDYT?